### ISSUE TITLE
Feature request -- nested object skipping

### ISSUE THREAD

**Issue by jrmolin** [camgunz/cmp#71]
After reading through the code (and using it in a pet project -- great work! Thanks!!), I have come across a situation where I need some way to skip over nested objects to find specific keys in a map. I have been unpacking into a nested map and then walking the map, but that adds a frustrating amount of complexity to my case, when I'd rather just use the msgpack binary blob as the memory backing the data structure.

You have in your comments that there is no way to do nested skips without allocations, but couldn't you just add a counter into `cmp_ctx_t` that tracks some limited amount of nesting (configurable at compile-time)?

```
#define CMP_RECURSION_DEPTH 4
#define CMP_RECURSION_DEPTH_BYTES 1

uint8_t recursion_count[CMP_RECURSION_DEPTH * CMP_RECURSION_DEPTH_BYTES];
```

Then each recursion can use `CMP_RECURSION_DEPTH_BYTES` to count up to the number of elements per recursion, and `CMP_RECURSION_DEPTH` can count up to the number of recursions supported. Obviously, this would be easier with allocations, but I completely understand your reluctance to add that.

Alternatively, you could handle it at runtime by adding

```
  uint8_t recursion_max; // set by cmp_ctx_init_with_recursion (new function)
  uint8_t recursion_stack[0]; // allocated along with the cmp_ctx_t struct
```
to the `cmp_ctx_t` struct. I think I'm over-complicating your very nice library, though. 

If this is something you think is worthwhile, I would be happy to work on it and put in a PR for review. If not, I will likely still work on it and just keep the code in my own project. Thoughts?

**Comment by m-wichmann** [comment]
Not sure if this solves your problem, but recursive skipping without allocation was implemented a couple of years ago (ea325ce). See `cmp_skip_object()`, `cmp_skip_object_limit()` and `cmp_skip_object_no_limit()`. With this you can e.g. skip over a map with multiple keys and values.

**Comment by jrmolin** [comment]
I read through the code, and it looked like the `skip` methods don't skip over the object, but actually read into the object for later skipping. I'll dig in again and see if that works for me. Thanks for responding!
